sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -442,6 +442,10 @@ case class InSubquery(values: Seq[Expression], query: ListQuery)
 // scalastyle:on line.size.limit
 case class In(value: Expression, list: Seq[Expression]) extends Predicate {
 
+  def this(valueAndList: Seq[Expression]) = {
```

Comment:
does it means we support `select in(a, b, c)` now?

Reply:
Yes. But it was not a goal, it is just a useful side effect.

Reply:
I'm not sure if this is useful. `in(a, b, c)` looks pretty weird to me. Can we revert it and treat `def in` as a special case? The SQL parser also treat `IN` as a special case and has dedicated syntax for it.

Reply:
All right, reverted in https://github.com/apache/spark/pull/42864/commits/8bf64a7f934a537ae00421c54a3b115a25c21168.